py/src/braintrust/logger.py

```diff
@@ -624,95 +810,71 @@ def log(
+        :param metrics: (Optional) a dictionary of metrics to log. The following keys are populated automatically and should not be specified: "start", "end", "caller_filename", "caller_lineno".
+        :param id: (Optional) a unique identifier for the event. If you don't provide one, BrainTrust will generate one for you.
+        :param dataset_record_id: (Optional) the id of the dataset record that this event is associated with. This field is required if and only if the experiment is associated with a dataset.
         :param inputs: (Deprecated) the same as `input` (will be removed in a future version)
         :returns: The `id` of the logged event.
         """
+        self._check_not_finished()
+
+        event = _validate_and_sanitize_experiment_log_full_args(
+            dict(
+                input=input,
+                output=output,
+                expected=expected,
+                scores=scores,
+                metadata=metadata,
+                metrics=metrics,
+                id=id,
+                dataset_record_id=dataset_record_id,
+                inputs=inputs,
+            ),
+            self.dataset is not None,
+        )
+        span = self.start_span(start_time=self.last_start_time, **event)
+        self.last_start_time = span.end()
+        return span.id
 
-        user_id = user_info()["id"]
-
-        if input is None and inputs is None:
-            raise ValueError("Either input or inputs (deprecated) must be specified. Prefer input.")
-        elif input is not None and inputs is not None:
-            raise ValueError("Only one of input or inputs (deprecated) can be specified. Prefer input.")
-
-        if not isinstance(scores, dict):
-            raise ValueError("scores must be a dictionary of names with scores")
-        for name, score in scores.items():
-            if not isinstance(name, str):
-                raise ValueError("score names must be strings")
-
-            if isinstance(score, bool):
-                score = 1 if score else 0
-                scores[name] = score
-
-            if score is not None:
-                if not isinstance(score, (int, float)):
-                    raise ValueError("score values must be numbers")
-                if score < 0 or score > 1:
-                    raise ValueError(f"score ({score}) values must be between 0 and 1")
-
-        if metadata:
-            if not isinstance(metadata, dict):
-                raise ValueError("metadata must be a dictionary")
-            for key in metadata.keys():
-                if not isinstance(key, str):
-                    raise ValueError("metadata keys must be strings")
+    def start_span(self, name="root", span_attributes={}, start_time=None, set_current=False, **event):
```

Comment:
can we make `set_current=True` the default :)

Reply:
Yes :)